### PR TITLE
[Authentication]: added AuthenticationController to provide registration

### PR DESCRIPTION
**Pull request by vadymhrnk**
## Changes:
- **`application.properties`** context path is now `"/api"`
## Added:
- MapperConfig
- AuthenticationController 
- RegistrationException 
- UserMapper
- UserRepository
- UserService

### CODE REVIEW

**Review by annton-shakur — APPROVED** [review]
lgtm

File: src/main/java/com/example/accommodationbookingapp/service/user/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto requestDto) {
+        if (userRepository.findByEmail(requestDto.getEmail()).isPresent()) {
+            throw new RegistrationException(
+                    "Can't register a new user with email: "
```
Comment (MinoDentori) [inline src/main/java/com/example/accommodationbookingapp/service/user/impl/UserServiceImpl.java:23]: Will we create constants for messages like this, or not?

**Review by MinoDentori — APPROVED** [review]
Looks god!

File: src/main/java/com/example/accommodationbookingapp/service/user/impl/UserServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+                            + requestDto.getEmail()
+            );
+        }
+        User user = userMapper.toModel(requestDto);
+        return userMapper.toDto(userRepository.save(user));
```
Comment (TarasYurkovskyi) [inline src/main/java/com/example/accommodationbookingapp/service/user/impl/UserServiceImpl.java:28]: i think u have to add this user.setPassword(passwordEncoder.encode(requestDto.getPassword()));

**Review by TarasYurkovskyi — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/accommodationbookingapp/service/user/impl/UserServiceImpl.java (hunk 2 above)
Comment (vadymhrnk) [reply src/main/java/com/example/accommodationbookingapp/service/user/impl/UserServiceImpl.java:28]: I will implement this in next PR with all the security config

Comment (TarasYurkovskyi) [reply src/main/java/com/example/accommodationbookingapp/service/user/impl/UserServiceImpl.java:28]: > I will implement this in next PR with all the security config

ok

**Review by TarasYurkovskyi — APPROVED** [review]
GJ!

**Pull request merged**